File: Lecture 1/dotnet_project/ReadParseFiles.cs
using System;
using YamlDotNet.Serialization;
using System.Text.Json;

namespace MyProgram
{


    public class ReadParseFiles
    {

        // File Info

        private readonly string fileFolder = "../files/";
        private readonly string textFilename = "text_file.txt";
        private readonly string jsonFilename = "json_file.json";
        private readonly string yamlFilename = "yaml_file.yaml";
        private readonly string csvFilename = "csv_file.csv";

        // Data Buckets

        private dynamic? txtContents;
        private dynamic? jsonContents;
        private dynamic? yamlContents;

        public ReadParseFiles()
        {

        }

        public void readParseTextFile()
        {

            string filepath = $"{fileFolder + textFilename}";

            try
            {
                txtContents = File.ReadAllText(filepath);
                //Console.WriteLine(txtContents);
            }
            catch (FileNotFoundException ex)
            {
                Console.WriteLine($"The file could not be found: {ex.Message}");
            }
            catch (Exception ex)
            {
                Console.WriteLine($"An error occurred while reading the file: {ex.Message}");
            }

        }

        public void readParseJsonFile()
        {

            string filepath = $"{fileFolder + jsonFilename}";

            try
            {
                var cache = File.ReadAllText(filepath);
                // converting the jsonString to a jsonObject
                jsonContents = JsonSerializer.Deserialize<JsonElement>(cache);
                // Console.WriteLine(jsonContents);
                // // Testing if object was successfully created
                // string name = jsonContents.GetProperty("name").GetString();
                // Console.WriteLine(name);
            }
            catch (FileNotFoundException ex)
            {
                Console.WriteLine($"The file could not be found: {ex.Message}");
            }
            catch (Exception ex)
            {
                Console.WriteLine($"An error occurred while reading the file: {ex.Message}");
            }

        }

        public void readParseYamlToJson()
        {

            string filepath = $"{fileFolder + yamlFilename}";

            try
            {
                var cache = File.ReadAllText(filepath);
                var deserializer = new DeserializerBuilder().Build();
                // Converting the string to a yaml object
                var yamlObject = deserializer.Deserialize(new StringReader(cache));
                // Converting the yamlObject to a jsonString
                var jsonString = JsonSerializer.Serialize(yamlObject, new JsonSerializerOptions { WriteIndented = true });
                // Converting the jsonString to a jsonObject
                yamlContents = JsonSerializer.Deserialize<JsonElement>(jsonString);
                // Console.WriteLine(yamlContents);
                // // Testing if object was successfully created
                // string name = yamlContents.GetProperty("name").GetString();
                // Console.WriteLine(name);
            }
            catch (FileNotFoundException ex)
            {
                Console.WriteLine($"The file could not be found: {ex.Message}");
            }
            catch (Exception ex)
            {
                Console.WriteLine($"An error occurred while reading the file: {ex.Message}");
            }

        }

    }

}

File: Lecture 1/Assignment/python_project/read_parse_files.py
import json
import csv
import xmltodict

class ReadParseFiles:

    def __init__(self):
        # Using name mangling to make the variable harder to access
        self.__fileFolder = "../files/"
        self.__textFileName = "text_file.txt"
        self.__jsonFileName = "json_file.json"
        self.__yamlFilename = "yaml_file.yaml"
        self.__csvFileName = "csv_file.csv"
        self.__xmlFileName = "xml_file.xml"
        self.__txtContents = None
        self.__jsonContents = None
        self.__yamlContents = None
        self.__csvContents = None
        self.__xmlContents = None

    def __call__(self):
        self.readParseTextFile()
        self.readParseJsonFile()
        self.readParseCsvToJson()

    def fileFolder(self):
        return self.__fileFolder
    
    def textFileName(self):
        return self.__textFileName
    
    def jsonFileName(self):
        return self.__jsonFileName
    
    def yamlFileName(self):
        return self.__yamlFilename
    
    def csvFileName(self):
        return self.__csvFileName
    
    def xmlFileName(self):
        return self.__xmlFileName
    

    def readParseTextFile(self):

        filepath = f"{self.fileFolder()}{self.textFileName()}"

        try:

            with open(filepath, 'r') as file:
                self.__txtContents = file.read()
                #print(self.__txtContents)
                
        except IOError as err:

            print(f"Error reading file: {err}")


    def readParseJsonFile(self):
        
        filepath = f"{self.fileFolder()}{self.jsonFileName()}"

        try:
            
            with open(filepath, 'r') as file:
                self.__jsonContents = json.load(file)

            jsonString = json.dumps(self.__jsonContents, indent=2)

            #print(jsonString)
            #print(self.__jsonContents["name"])

        except IOError as err:
            print(f"Error reading file: {err}")
    

    def readParseCsvToJson(self):

        filepath = f"{self.fileFolder()}{self.csvFileName()}"

        try:

            with open(filepath, 'r') as file:
                reader = csv.DictReader(file)
                csvData = [row for row in reader]

            self.csvContents = [self.transformCsvToPerson(row) for row in csvData]

            jsonString = json.dumps(self.csvContents, indent=2)
            self.__csvContents = json.loads(jsonString)
            selectedJsonElement = self.__csvContents[0]
            formattetElement = json.dumps(selectedJsonElement, indent=2)
            print(formattetElement)
            print(selectedJsonElement["name"])

        except IOError as err:

            print(f"Error reading file: {err}")


    def transformCsvToPerson(self, row):
            name_parts = row["name"].split(" ")
            return {
                "name" : f"{name_parts[0]} {name_parts[1]}",
                "age" : row["age"],
                "email" : row["email"],
                "address": {
                    "street": row["street"],
                    "city": row["city"],
                    "state": row["state"],
                    "zip": row["zip"]
                }
            }
    
    def serialize_xml_to_json(self):

        filepath = f"{self.fileFolder()}{self.xmlFileName()}"
        jsonObject = None

        try:

            with open(filepath, 'r') as file:
                xml_data = file.read()
            
            # Serialing the xml_date to jsonString
            jsonString = json.dumps(xmltodict.parse(xml_data), indent=2)

            # It is required by FastAPI to parse/marshal the jsonString to an Json Object
            # before sending it as a response.
            jsonObject = json.loads(jsonString)
        
        except IOError as err:

            print(f"{err}")

        return jsonObject